src/backends/torch/torchsolver.cc

```diff
@@ -306,30 +333,15 @@ namespace dd
                 _logger->info("no previous best model file");
               }
           }
-        if (mlmodel._sstate.empty())
-          {
-            throw MLLibBadParamException(
-                "resuming a model requires a solverstate file in model "
-                "repository");
-          }
-        else
-          try
-            {
-              return load(mlmodel._sstate, main_device);
-            }
-          catch (std::exception &e)
-            {
-              this->_logger->error("Failed to load solver state");
-              throw;
-            }
       }
     else if (!mlmodel._sstate.empty())
       {
         this->_logger->error("not resuming while a solverstate file remains "
                              "in model repository");
         throw MLLibBadParamException(
             "a solverstate file requires a resume argument for training, "
```

Comment:
Same remark as above, we should swap `solverstate` with another formulation that is better grounded to torch filenames.